### PR TITLE
add string array to learn

### PR DESCRIPTION
**Pull request by freevryheid**

### CODE REVIEW

**Comment by freevryheid** [comment]
Update the example to show how this string array is passed to another function.

**Review by awvwgk — APPROVED** [review]
Thanks for sharing. This is a useful addition to the learn section. Beside one question, this looks good to me.

- should we trim the strings in the printout to show how to deal with trailing whitespace?

File: learn/quickstart/arrays_strings.md [hunk 1]
```diff
@@ -149,3 +149,34 @@ program allocatable_string
 end program allocatable_string
 ```
 
+
+## Array of strings
+
+The length of all strings in the array must be the same. The length assigned is typically the length of the longest element.
```
Comment (milancurcic) [inline learn/quickstart/arrays_strings.md:155]: I thought the explanation needed some clarification, I hope you like it.

```suggestion
An array of strings can be expressed in Fortran as an array of `character` variables.
All elements in a `character` array have equal length.
However, strings of varying lengths can be provided as input to the array constructor, as shown in the example below.
They will be truncated or right-padded with spaces if they are longer or shorter, respectively, than the declared length of the `character` array.
If the `character` array is declared as deferred-length (`character(len=:), allocatable`) instead, the `character` array will be allocated with the length of the longest element.
```

File: learn/quickstart/arrays_strings.md [hunk 2]
```diff
@@ -149,3 +149,34 @@ program allocatable_string
...
+
+
+__Example:__ string array
+```fortran
+program string_array
+  implicit none
+  character(len=10), dimension(2) :: keys, vals
+
+  keys = [character(len=10)        :: "user", "dbname"]
+  vals = [character(len=10)        :: "ben", "motivation"]
```
Comment (milancurcic) [inline learn/quickstart/arrays_strings.md:165]: Minor style change

```suggestion
  keys = [character(len=10) :: "user", "dbname"]
  vals = [character(len=10) :: "ben", "motivation"]
```

File: learn/quickstart/arrays_strings.md [hunk 3]
```diff
@@ -149,3 +149,34 @@ program allocatable_string
...
+
+    subroutine show(akeys, avals)
+      character(len=*), intent(in) :: akeys(:), avals(:)
+      integer                      :: i
+
+      do i = 1, size(akeys)
+        print *, akeys(i), ": ", avals(i)
+      end do
+
+    end subroutine show
```
Comment (milancurcic) [inline learn/quickstart/arrays_strings.md:179]: I thought the indentation of `contains` was unnecessary, let's keep indentation at a minimum for better readability on mobile.

```suggestion
contains

  subroutine show(akeys, avals)
    character(len=*), intent(in) :: akeys(:), avals(:)
    integer                      :: i

    do i = 1, size(akeys)
      print *, akeys(i), ": ", avals(i)
    end do

  end subroutine show
```

**Review by milancurcic — CHANGES_REQUESTED** [review]
Thank you, great addition. I left a few suggestions.

File: learn/quickstart/arrays_strings.md (hunk 1 above)
Comment (freevryheid) [reply learn/quickstart/arrays_strings.md:155]: Sorry, noted a spelling error after resolving - line 158, They will be truncated ... th(a)n the declared length ...

Comment (milancurcic) [reply learn/quickstart/arrays_strings.md:155]: Good catch, thanks. I noticed that you resolved the suggestions without committing them. Did you mean to do that?

File: learn/quickstart/arrays_strings.md [hunk 4]
```diff
@@ -149,3 +149,37 @@ program allocatable_string
...
+  call show(keys, vals)
+
+  contains
+
+  subroutine show(akeys, avals)
+    character(len=*), intent(in) :: akeys(:), avals(:)
+    integer                      :: i
+
+    do i = 1, size(akeys)
+      print *, akeys(i), ": ", avals(i)
```
Comment (milancurcic) [inline learn/quickstart/arrays_strings.md:179]:
```suggestion
      print *, trim(akeys(i)), ": ", trim(avals(i))
```

**Review by milancurcic — APPROVED** [review]
Implemented the suggestion by Sebastian.

**Comment by milancurcic** [comment]
Great, I'll go ahead and merge, thank you again.

**Pull request merged**